bicep/2025-06-manifest-to-bicep-extension-cli.md

```diff
@@ -0,0 +1,193 @@
+2. **Local Storage**: The binary is stored in `~/.rad/bin/manifest-to-bicep-extension` (with `.exe` extension on Windows)
+3. **Platform Support**: Binaries are available for the same platforms as Bicep (Windows, Linux, macOS on AMD64 and ARM64)
+
+#### Changes to rad bicep download
+
+The `rad bicep download` command now downloads both binaries:
+- The existing Bicep CLI download logic remains unchanged
+- After downloading Bicep, it also downloads the manifest-to-bicep-extension binary
+- Both downloads use the same retry mechanism and error handling
+
+#### Changes to rad bicep publish-extension
+
+The command now:
+1. Checks if the manifest-to-bicep-extension binary exists
+2. If not found, prompts the user to run `rad bicep download`
+3. Executes the binary directly instead of using `npx`
+
+### Implementation Details
+
+#### CLI Updates
+
+The key changes are in the following files:
+
+1. **pkg/cli/bicep/bicep.go**: Added `GetBicepManifestToBicepExtensionCLIPath()` function and updated `DownloadBicep()` to download both binaries
+
+2. **pkg/cli/bicep/tools/download_tools.go**: Added `DownloadManifestToBicepExtensionCLI()` function
+   - Downloads the manifest-to-bicep-extension binary from GitHub releases
+   - Handles platform-specific naming and paths
+
+3. **pkg/cli/cmd/bicep/publishextension/publish.go**: Updated to use the binary instead of `npx`
+   - Checks for the binary's existence
+   - Executes the binary with the provided arguments
+
+#### Build System Updates
+
+1. **GitHub Actions**: Updated workflows to install and use the binary instead of `npx`
+2. Updated `radius-project/bicep-tools` repository to include the manifest-to-bicep-extension binary in releases
+
+### Error Handling
+
+- If the manifest-to-bicep-extension binary is not found, the user is prompted to run `rad bicep download`
+- Download failures are retried using the existing retry mechanism
+- Platform-specific errors (unsupported OS/architecture) provide clear error messages
+
+## Test plan
+
+1. **Unit Tests**:
+   - Test binary path resolution
+   - Test platform detection logic
+   - Test error handling when binary is missing
+
+2. **Functional Tests**:
+   - Remove `npx` CLI install from test workflows
+   - Use existing test cases to verify that `rad bicep publish-extension` works with the new binary
+
+## Security
+
+- The binary is downloaded over HTTPS from GitHub releases
+- The binary is signed and verified (same as Bicep CLI)
+- No new security considerations beyond existing binary management
+
+## Compatibility
+
+- This is a breaking change for users who rely on the NPX-based approach
+- Users must run `rad bicep download` to get the new binary
+- The command-line interface remains unchanged
+
+## Monitoring and Logging
+
+- Download progress and errors are logged to the console
+- Binary execution errors are propagated to the user
+- No new telemetry is added
+
+## Development plan
```

Comment:
Do we need a step to modify and test the release process to ensure that the binary is available, as described on line 89?